demos/context.py

```diff
@@ -1,8 +1,7 @@
-import sys
 import os
+import sys

```

Comment:
I am genuinely curious if there is a rule/best practice for the order of these system imports?

Reply:
In my experience, it is that they occur at the top of imports and are alphabetized within their import group

Reply:
I think the only one that most linters apply universally is alphabetical ordering. On my own I was sort of applying the following practice (which I picked up at my first job):
```
# project `from` imports first
from palimpzest.* import *

# project direct imports second
import palimpzest.* as *

# external from imports third
from abc import *

# external direct imports fourth
import pandas as pd 
```
But it looks like Ruff is just doing alphabetical, and `import` before `from`

Reply:
Ruff is doing import groups and sorting within the groups. It is not just strictly alphabetical